src/common/util.js

```diff
@@ -144,13 +180,26 @@ function skipNewline(text, index, opts) {
   return index;
 }
 
+/**
+ * @param {string} text
+ * @param {number | false} index
```

Comment:
Is this function really supposed to take `number | false` as input? Is anyone calling it with `false`?

Reply:
As a human I would say: no
As a compiler I would say: yes, because L251 (`isNextLineEmptyAfterIndex`) is calling it with `idx: number | false`
And it uses many of the `skip*` functions

Reply:
Then it sounds to me like L251 should be:

```js
return idx === false ? false : hasNewline(text, idx)
```

And that `hasNewline` should only accept `number` for `idx`.